klippy/chelper/trapq.c

```diff
@@ -190,10 +189,9 @@ trapq_finalize_moves(struct trapq *tq, double print_time)
     if (list_empty(&tq->history))
         return;
     struct move *latest = list_first_entry(&tq->history, struct move, node);
-    double expire_time = latest->print_time + latest->move_t - HISTORY_EXPIRE;
```

Comment:
Just for the record I think we're now clearing the history in places where we wouldn't have before which is a change in behavior. When `print_time` is `NEVER_TIME` (or a large value) all the moves would get transferred to the history, but this code would still only clear 30s prior to the last move because of the math on this line. This is why I had the new parameter as 0 initially (to keep the history until this is called from the toolhead code).

Reply:
Okay, I goofed.  You are correct and I misread the original code.  I think we do want to avoid a change in behavior here, because this is an area of the code where we really want to be able to use tools like `git bisect` if necessary.

If I'm reading the code correctly, the call in `trapq_finalize_moves()` doesn't need to flush the history (it can use `0`), and the code in `DripModeEndSignal` doesn't need to flush (it can also use `0`).  However, the code in manual_stepper.py and force_move.py do need to fully flush (using the current `+ 99999.9` method) as nothing else will free the history (and nothing uses the history of these modules).

Does that sound correct?

Thanks.
-Kevin

Reply:
If you mean `trapq_set_position` and not `trapq_finalize_moves` that looks correct to me (as in, it preserves the current behavior), however I'm not super familiar with this code. I left the other 2 calls with `+ 99999.9` to clear the history.